### PR TITLE
Add AutoScaling Groups

### PR DESCRIPTION
**Pull request by Nss**
Part of https://github.com/nearform/nscale-kernel/pull/101

### CODE REVIEW

**Comment by mcollina** [comment]
You should also add an aws-autoscaling-machine for each instance attached to the autoscaling group.

**Comment by Nss** [comment]
You are right, there wasn't any machine while I was developing this, I'll add them ;)

**Comment by Nss** [comment]
@mcollina check it now

File: lib/ec2-autoscaling-groups.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+var _ = require('lodash');
+
+/**
+ * analyze the AWS AutoScalingGroups. The following tags must be in place on the Group
+ *
+ * nscale-system: <namespace>-<target> for example nodezoo-blue
+ * Name: <nscaleId> for example awsMachine$fe-12345
+ */
+module.exports = function fetchAutoScalingGroups(config, result, cb) {
+  var ec2 = new AWS.AutoScaling(config);
```
Comment (mcollina) [inline lib/ec2-autoscaling-groups.js]: don't call this object `ec2`, but `autoscaling`, so we know what it is.

Comment (Nss) [inline lib/ec2-autoscaling-groups.js]: Wow nice catch, I forgot about it

File: lib/ec2-autoscaling-groups.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          containedBy: autoScalingGroup.LoadBalancerNames,
+          containerDefinitionId: defName,
+          type: 'aws-autoscaling',
+          contains: _.pluck(instances, 'InstanceId'),
+          specific: autoScalingGroup,
+          nativeId: autoScalingGroup.AutoScalingGroupName
+        };
+
+        container = _.find(containers, {contains: [id]});
+        console.log(container);
```
Comment (mcollina) [inline lib/ec2-autoscaling-groups.js]: console.log :(

**Comment by mcollina** [comment]
This is great!

**Comment by Nss** [comment]
This should fix the problem with the missing relation of awsSg in the analyzer's result tree, again @mcollina review when you can

**Comment by mcollina** [comment]
Seems good, I will test it this afternoon.

**Comment by mcollina** [comment]
good for me. Still a minor issue on the docker ids remains.

**Pull request merged**